### PR TITLE
feat: Add support for White Label features

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add support for White Label features
- Add unit tests for White Label features (disable credit card test due to VCR)
- Add new fixtures as needed

# Testing
- New unit tests added and recorded
- Unit test for adding credit card disabled due to VCR recording issue
  - Manually confirmed works via E2E test
  
# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

File: referral.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+
+		}
```
Comment (Justintime50) [inline referral.go]: Question: Shouldn't we do something with this?

File: Makefile [hunk 2]
```diff
@@ -21,6 +21,7 @@ coverage:
 
 ## install - Install and vendor dependencies
 install:
+	brew install golangci-lint
```
Comment (Justintime50) [inline Makefile:24]: This needs to `exit 0` as the fallback incase it's already installed.

File: referral.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Println(err) // we should never get here
```
Comment (Justintime50) [inline referral.go:150]: Issue: We don't use the `log` package in this project, let's not introduce it here. Check out other places where we raise errors.

File: tests/cassettes/TestReferralCustomerAddCreditCard.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (Justintime50) [inline tests/cassettes/TestReferralCustomerAddCreditCard.yaml:1]: Delete this cassette since it won't be used

File: Makefile (hunk 2 above)
Comment (nwithan8) [reply Makefile:24]: Fixed

File: referral.go (hunk 3 above)
Comment (nwithan8) [reply referral.go:150]: Resolved

File: tests/cassettes/TestReferralCustomerAddCreditCard.yaml (hunk 4 above)
Comment (nwithan8) [reply tests/cassettes/TestReferralCustomerAddCreditCard.yaml:1]: Resolved

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**